src/core/mesh/qgsmeshdataprovider.h

```diff
@@ -168,6 +168,11 @@ class CORE_EXPORT QgsMeshDatasetGroupMetadata
      */
     bool isOnVertices() const;
 
+    /**
+     * \brief Returns whether dataset group data is defined on faces
+     */
+    bool isOnFaces() const;
```

Comment:
Maybe it would be nicer to replace isOnVertices() / isOnFaces() with a "type" enum. It would be easier to understand that "on vertices" and "on faces" are mutually exclusive